### DISCUSSION TITLE
Separate MCU Banks per potentiometer over ManyAddresses

### DISCUSSION THREAD

**Discussion by gringow831** [tttapa/Control-Surface#453] [General]
Hi Pieter,

I was wondering if there is a way of using MCU Bank per potentiometer with ManyAddresses, changing the MCU Bank depending on needs with the same two buttons rather than using a different cable. Otherwise I'll have to use a Teensy to create a MackieControlXT with Cable over ManyAddresses

I haven't written the code just yet but this is an idea of what I am trying to figure out.

Bankable::ManyAddresses::PBPotentiometer<3> knobs[] = {
{ bank, A0, {MCU::VOLUME_1, MCU::VOLUME_1 (BANK_2), MCU::VOLUME_5,}},
{ bank, A1, {MCU::VOLUME_2, MCU::VOLUME_2 (BANK_2), MCU::VOLUME_6,}},
{ bank, A2, {MCU::VOLUME_3, MCU::VOLUME_3 (BANK_2), MCU::VOLUME_7,}},
{ bank, A3, {MCU::VOLUME_4, MCU::VOLUME_4 (BANK_2), MCU::VOLUME_8,}},
{ bank, 6, {MCU::VOLUME_5, MCU::VOLUME_5 (BANK_2), MCU::VOLUME_1 (BANK_2),}},
};

At the moment by pressing the bank button all potentiometers change at the same time to the second bank and I tries separating them but didn't work. So the 5th pot would go back to Volume_5 by pressing the bank rather than going to VOLUME_1 on the seconds bank.
The 5th Potentiometer goes to volume_1 on the third instance while staying on bank two meanwhile the rest goes to Bank once again.

Can't find the code I wrote to try and it didn't work but it was something similar to the following.

Bankable::ManyAddresses::PBPotentiometer<3> knobs[] = {
{ bank, A0, {MCU::VOLUME_1, MCU::VOLUME_1 (BANK_2), MCU::VOLUME_5,}},
};

Bankable::ManyAddresses::PBPotentiometer<3> knobs2[] = {
{ bank, A0, {MCU::VOLUME_5, MCU::VOLUME_5 (BANK_2), MCU::VOLUME_1 (BANK_2),}},
};

Thanks
Guido

**Comment by tttapa** [comment]
It's not entirely clear to me what you're trying to do. Are you using the `MCU::BANK_LEFT/RIGHT` commands?
